ui/packages/consul-ui/app/locations/fsm-with-optional.js

```diff
@@ -190,9 +190,9 @@ export default class FSMWithOptionalLocation {
     if (typeof hash.nspace !== 'undefined') {
       hash.nspace = `~${hash.nspace}`;
     }
-    // if (typeof hash.partition !== 'undefined') {
-    //   hash.partition = `-${hash.partition}`;
-    // }
+    if (typeof hash.partition !== 'undefined') {
+      hash.partition = `-${hash.partition}`;
+    }
```

Comment:
These two changes allow `/-partition-name/` in URLs (both on the way in and out)